<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HIDMessageAnalyserClass</class>
 <widget class="QMainWindow" name="HIDMessageAnalyserClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>982</width>
    <height>326</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>982</width>
    <height>326</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>982</width>
    <height>326</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>HIDMessageAnalyser</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>982</width>
        <height>326</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>982</width>
        <height>326</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_make">
       <attribute name="title">
        <string>生成HID消息</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>选择数据类型</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QRadioButton" name="rdb_random_send">
             <property name="text">
              <string>随机写入</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rdb_random_read">
             <property name="text">
              <string>随机读取</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rad_conti_send">
             <property name="text">
              <string>连续写入</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rdb_conti_read">
             <property name="text">
              <string>连续读取</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="minimumSize">
           <size>
            <width>450</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>参数</string>
          </property>
          <widget class="QStackedWidget" name="widgets">
           <property name="geometry">
            <rect>
             <x>1</x>
             <y>13</y>
             <width>450</width>
             <height>231</height>
            </rect>
           </property>
           <property name="currentIndex">
            <number>2</number>
           </property>
           <widget class="QWidget" name="wid_random_write">
            <widget class="QLabel" name="label">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>101</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>随机写入地址位：</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_3">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>56</y>
               <width>36</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>参数：</string>
             </property>
            </widget>
            <widget class="QPushButton" name="btn_r_w">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>140</y>
               <width>161</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>计算随机写入命令</string>
             </property>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>74</y>
               <width>181</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLineEdit" name="line_r_w_sval"/>
              </item>
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>例:1</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>28</y>
               <width>183</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLineEdit" name="line_r_w_addr"/>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>例:082A</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
           <widget class="QWidget" name="wid_random_read">
            <widget class="QLabel" name="label_6">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>111</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>随机读取地址位：</string>
             </property>
            </widget>
            <widget class="QPushButton" name="btn_r_r">
             <property name="geometry">
              <rect>
               <x>40</x>
               <y>180</y>
               <width>151</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>计算随机读取命令</string>
             </property>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>40</y>
               <width>191</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLineEdit" name="line_r_r_addr"/>
              </item>
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>例:082A</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
           <widget class="QWidget" name="wid_conti_write">
            <widget class="QLabel" name="label_8">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>101</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>连续写入地址:</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_10">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>70</y>
               <width>181</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>连续写入内容（使用|分割):</string>
             </property>
            </widget>
            <widget class="QPushButton" name="btn_c_w">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>150</y>
               <width>141</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>计算连续写入命令</string>
             </property>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>40</y>
               <width>191</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLineEdit" name="line_c_w_addr"/>
              </item>
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>例:082A</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>100</y>
               <width>321</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLineEdit" name="line_c_w_sval"/>
              </item>
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>例:1|32A8|8AB4</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
           <widget class="QWidget" name="wid_conti_read">
            <widget class="QLabel" name="label_12">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>10</y>
               <width>131</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>连续参数读取地址：</string>
             </property>
            </widget>
            <widget class="QLabel" name="label_14">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>80</y>
               <width>111</width>
               <height>16</height>
              </rect>
             </property>
             <property name="text">
              <string>连续参数读取长度：</string>
             </property>
            </widget>
            <widget class="QPushButton" name="btn_c_r">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>170</y>
               <width>181</width>
               <height>23</height>
              </rect>
             </property>
             <property name="text">
              <string>计算连续参数读取命令</string>
             </property>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>40</y>
               <width>241</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLineEdit" name="line_c_r_addr"/>
              </item>
              <item>
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>例:082A</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="layoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>110</y>
               <width>241</width>
               <height>22</height>
              </rect>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLineEdit" name="lline_c_r_sval">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_15">
                <property name="text">
                 <string>例如:12</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>计算结果：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="text">
            <property name="minimumSize">
             <size>
              <width>400</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>400</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_analyse">
       <attribute name="title">
        <string>解析HID消息</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>输入参数 例: c40b0000037e204a0046fe</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="text_input"/>
          </item>
          <item>
           <widget class="QPushButton" name="btn_analyse">
            <property name="text">
             <string>解析输入</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="minimumSize">
           <size>
            <width>700</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>700</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>解析结果：</string>
          </property>
          <widget class="QLabel" name="label_17">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>61</width>
             <height>16</height>
            </rect>
           </property>
           <property name="text">
            <string>参数类型:</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="line_ret_addr_type">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>40</y>
             <width>113</width>
             <height>20</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_18">
           <property name="geometry">
            <rect>
             <x>140</x>
             <y>20</y>
             <width>61</width>
             <height>16</height>
            </rect>
           </property>
           <property name="text">
            <string>参数地址:</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="line_ret_addr">
           <property name="geometry">
            <rect>
             <x>140</x>
             <y>40</y>
             <width>71</width>
             <height>20</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_19">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>70</y>
             <width>61</width>
             <height>16</height>
            </rect>
           </property>
           <property name="text">
            <string>读取参数:</string>
           </property>
          </widget>
          <widget class="QTableWidget" name="tablewidget_ret">
           <property name="geometry">
            <rect>
             <x>9</x>
             <y>90</y>
             <width>201</width>
             <height>171</height>
            </rect>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAsNeeded</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <column>
            <property name="text">
             <string>地址</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>参数</string>
            </property>
           </column>
          </widget>
          <widget class="QLabel" name="label_20">
           <property name="geometry">
            <rect>
             <x>220</x>
             <y>20</y>
             <width>54</width>
             <height>12</height>
            </rect>
           </property>
           <property name="text">
            <string>结果：</string>
           </property>
          </widget>
          <widget class="QPlainTextEdit" name="text_analyse">
           <property name="geometry">
            <rect>
             <x>220</x>
             <y>40</y>
             <width>261</width>
             <height>241</height>
            </rect>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="HIDMessageAnalyser.qrc"/>
 </resources>
 <connections/>
</ui>
